makelib/common.mk

```diff
@@ -300,7 +300,6 @@ do.build.artifacts: $(foreach p,$(PLATFORMS), do.build.artifacts.$(p))
 # build for all platforms
 build.all:
 	@$(MAKE) build.init
-	@$(MAKE) build.check
```

Comment:
The most notable change here is that this line is removed. As I mentioned before this is really debatable. It's quite nice to have it executed as part of the build process, but at the same time the CI is in place and does that, separately on different job.

Reply:
One thing to note here is that this build submodule is used by several Upbound internal projects, so these changes could affect more than just Crossplane. :( I think we eventually should fork the build library under the Crossplane org so that folks without visibility into Upbound's private repos can make changes safely.

This was mentioned in https://github.com/crossplane/crossplane/issues/1583, which I accidentally closed while only half done.

Reply:
That's a really valid point! Maybe as an interim workaround (before forking this under `crossplane` org) an environment variable can be added to disable this.

```makefile
RUN_BUILD_CHECK ?=true
build.all:
	@$(MAKE) build.init
ifeq ($(RUN_BUILD_CHECK),"true")
	@$(MAKE) build.check
endif
```